### PR TITLE
Use static_cast<int> inside each call to std::abs() in gplates.cc

### LINKED ISSUE #746: Call to std::abs() is ambiguous in GPlates::update() under clang 7.0.2 in Xcode 7.2
The new `GPlates::update()` commit https://github.com/geodynamics/aspect/commit/fb5b371f0b4f90389d50dfb1222a1d31c718e735 doesn't compile in Xcode.

Lines 654 and 682 in `source/velocity_boundary_conditions/gplates.cc` give the error `Call to abs() is ambiguous` and suggests candidate functions `abs(int)` in `/usr/include/stdlib.h`, `abs( long __x)` and `abs( long long __x)` in `cstdlib`, and `abs(float __lcpp_x)`, `abs(double __lcpp_x)`, and `abs(long double __lcpp_x)` in `cmath`.

Google suggests this is a known conflict between the headers, but I couldn't work out a immediate fix, as I expect several of these headers may already be in use in the include chain.

Apple LLVM version 7.0.2 (clang-700.1.81)
Target: x86_64-apple-darwin15.3.0
Thread model: posix

OSX El Capitan 10.11.3 (15D21)

Xcode 7.2

### PR DESCRIPTION
**Pull request by spco**
Fixes #746

### CODE REVIEW

File: source/velocity_boundary_conditions/gplates.cc [hunk 1]
```diff
@@ -673,13 +673,13 @@ namespace aspect
                 first_data_file_number
                 + static_cast<unsigned int> (time_steps_since_start);
 
-              const bool load_both_files = std::abs(current_file_number - old_file_number) >= 1;
+              const bool load_both_files = std::abs(static_cast<int>(current_file_number - old_file_number)) >= 1;
```
Comment (tjhei) [inline source/velocity_boundary_conditions/gplates.cc]: So ``current_file_number`` is an ``unsigned int`` and ``old_file_number`` is an ``int``. If you subtract them they are both converted to an unsigned first and might wrap around, right? I am not sure this is intended.

Comment (gassmoeller) [inline source/velocity_boundary_conditions/gplates.cc]: I am not sure this way everything still works as intended. There are just too many special cases (files starting from x > 0 or from 0, indices increasing with time or decreasing). I think this problem arises from my initial assumption that file numbers may not be negative, but thinking about it now, it seems that in fact they could. Somebody could name their files file_001, file_000, file_-001 to indicate this data is prescribing "the future" or "something before my plate motion model starts". So maybe the simplest solution is to do not make the `unsigned` assumption at all and handle everything as `int`. I can prepare a patch for this pull request later today (I think I know all the places this should affect).

**Comment by gassmoeller** [comment]
If this works for @spco, this PR should be ready for review and merge.

**Comment by spco** [comment]
Works fine for me

File: source/velocity_boundary_conditions/gplates.cc [hunk 2]
```diff
@@ -659,26 +660,26 @@ namespace aspect
...
-                (decreasing_file_order) ?
-                first_data_file_number
-                - static_cast<unsigned int> (time_steps_since_start)
+                (decreasing_file_order)
+                ?
+                first_data_file_number - time_steps_since_start
                 :
-                first_data_file_number
-                + static_cast<unsigned int> (time_steps_since_start);
+                first_data_file_number + time_steps_since_start;
```
Comment (tjhei) [inline source/velocity_boundary_conditions/gplates.cc]: can you make the order of operations between ? and +/- more clear with parentheses?

File: source/velocity_boundary_conditions/gplates.cc [hunk 3]
```diff
@@ -642,16 +642,17 @@ namespace aspect
...
     GPlates<dim>::update ()
     {
-      if (time_dependent && (this->get_time() - first_data_file_model_time >= 0.0))
+      const double time_since_start = this->get_time() - first_data_file_model_time;
+
+      if (time_dependent && (time_since_start >= 0.0))
         {
-          const double time_steps_since_start = (this->get_time() - first_data_file_model_time)
-                                                / data_file_time_step;
+          const int time_steps_since_start = static_cast<int> (time_since_start / data_file_time_step);
```
Comment (tjhei) [inline source/velocity_boundary_conditions/gplates.cc]: does this need rounding down?

Comment (bangerth) [inline source/velocity_boundary_conditions/gplates.cc]: `static_cast<int>` rounds towards zero, not towards the closest integer.

Comment (gassmoeller) [inline source/velocity_boundary_conditions/gplates.cc]: And the rounding to zero is intended in this case. I need the file number before the current time.

**Comment by gassmoeller** [comment]
When my changes have been merged into this PR by @spco this should be ready for merge.

**Comment by spco** [comment]
Compiles for me.

**Pull request merged**